lib/puppet/util/log.rb

```diff
@@ -18,11 +18,11 @@ class Puppet::Util::Log
   @desttypes = {}
 
   # Create a new destination type.
-  def self.newdesttype(name, options = {}, &block)
+  def self.newdesttype(name, options = {}, parent = nil, &block)
```

Comment:
Yes, it's backward compatible. The only dest_type that make use of this argument at present is the _json_file_ (passes the _file_ as it's parent). All others remain unchanged.